kart/cli.py

```diff
@@ -233,6 +246,210 @@ def push(ctx, do_progress, args):
     )
 
 
+@cli.command(context_settings=dict(ignore_unknown_options=True))
+@click.pass_context
+@click.option(
+    "--socket",
+    "socket_filename",
+    default=Path.home() / ".kart.socket",
+    show_default=True,
+    help="What socket to use",
+)
+@click.option(
+    "--timeout",
+    "timeout",
+    default=300,
+    show_default=True,
+    help="Timeout and shutdown helper when no commands received with this time",
+)
+@click.argument("args", nargs=-1, type=click.UNPROCESSED)
+def helper(ctx, socket_filename, timeout, args):
+    """Start the background helper process to speed up interaction"""
+    if is_windows:
+        click.echo("Helper mode not currently supported on Windows")
+        ctx.exit(1)
+
+    load_commands_from_args(["--help"])
+
+    # stash the required environment for used libs
+    # TODO - this should be checked to ensure it is all that is needed
+    #  is there anything beyond these which is needed, eg. PWD, USER, etc.
+    required_environment = {
+        k: v
+        for k, v in os.environ.items()
+        if k
+        in [
+            "PATH",
```

Comment:
Is there a reason we don't copy the entire environment - is it too big?

Reply:
The thinking there is to ensure that the environment that the helper is run in after it forks matches the environment for the 'required' values that it had when the original instance was run - library locations etc. I'd be happy to just pass everything from the client but we need combine that with those required libraries which are set in __init__.py and __init__.py isn't run in the forked helper.